src/neo/Oracle/OracleResult.cs

```diff
@@ -0,0 +1,155 @@
+        {
+            get => throw new NotImplementedException();
+            set => throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Create error result
+        /// </summary>
+        /// <param name="txHash">Tx Hash</param>
+        /// <param name="requestHash">Request Id</param>
+        /// <returns>OracleResult</returns>
+        public static OracleResult CreateError(UInt256 txHash, UInt160 requestHash, OracleResultError error)
+        {
+            return new OracleResult()
+            {
+                TransactionHash = txHash,
+                RequestHash = requestHash,
+                Error = error,
+                Result = new byte[0],
+            };
+        }
+
+        /// <summary>
+        /// Create result
+        /// </summary>
+        /// <param name="txHash">Tx Hash</param>
+        /// <param name="requestHash">Request Hash</param>
+        /// <param name="result">Result</param>
+        /// <returns>OracleResult</returns>
+        public static OracleResult CreateResult(UInt256 txHash, UInt160 requestHash, string result)
+        {
+            return CreateResult(txHash, requestHash, Encoding.UTF8.GetBytes(result));
+        }
+
+        /// <summary>
+        /// Create result
+        /// </summary>
+        /// <param name="txHash">Tx Hash</param>
+        /// <param name="requestHash">Request Id</param>
+        /// <param name="result">Result</param>
+        /// <returns>OracleResult</returns>
+        public static OracleResult CreateResult(UInt256 txHash, UInt160 requestHash, byte[] result)
+        {
+            return new OracleResult()
+            {
+                TransactionHash = txHash,
+                RequestHash = requestHash,
+                Error = OracleResultError.None,
+                Result = result,
+            };
+        }
+
+        public void SerializeUnsigned(BinaryWriter writer)
+        {
+            writer.Write(TransactionHash);
+            writer.Write(RequestHash);
+            writer.Write((byte)Error);
+            if (Error == OracleResultError.None)
+                writer.WriteVarBytes(Result);
+        }
+
+        public void Serialize(BinaryWriter writer)
+        {
+            SerializeUnsigned(writer);
+        }
+
+        public void DeserializeUnsigned(BinaryReader reader)
+        {
+            TransactionHash = reader.ReadSerializable<UInt256>();
+            RequestHash = reader.ReadSerializable<UInt160>();
+            Error = (OracleResultError)reader.ReadByte();
+            Result = Error != OracleResultError.None ? reader.ReadVarBytes(ushort.MaxValue) : new byte[0];
+        }
+
+        public void Deserialize(BinaryReader reader)
+        {
+            DeserializeUnsigned(reader);
+        }
+
+        public UInt160[] GetScriptHashesForVerifying(StoreView snapshot)
+        {
+            return new UInt160[] { new UInt160(Crypto.Hash160(this.GetHashData())) };
+        }
```

Comment:
Does not complete here, right?  Should we change it to `OracleConsensus`?

Reply:
I think that now it's only the result, when we finish the Consensus, we can change it according to the requirements.